### ISSUE TITLE
Cross-protocol redirect reuses same agent, but for instance http.Agent doesn't support followed https request

### ISSUE THREAD

**Issue by tenbits** [bitinn/node-fetch#571]
The same agent is reused on each followed redirect, but often the redirect crosses the protocol boundaries,  and it can cause an exception, when the agent can't handle the new protocol.

Simple example:

```
let agent = new http.Agent({ keepAlive: true });
let response = await fetch('http://google.com', { agent });
```

> TypeError [ERR_INVALID_PROTOCOL]: Protocol "https:" not supported. Expected "http:"

The reason is - the `node-fetch` follows the redirect to `https`, but still uses `http` agent. [index.js#L151](https://github.com/bitinn/node-fetch/blob/e996bdab73baf996cf2dbf25643c8fe2698c3249/src/index.js#L151)
As a consumer, we don't always know if and where the request will be redirected, so there should be some way to prevent agent forwarding on protocol boundaries, or some hooks should be implemented to give the consumer the ability to set another agent.

**Comment by bitinn** [comment]
Yes this is a known limit, but since agent is non-standard I doubt there will be much change to this.

However, if you are lucky enough to be using nodejs 11.7.0+, globalAgent is now replaceable, meaning you can control this precisely:

- https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V11.md#2019-01-17-version-1170-current-bridgear
- nodejs/node#25170

**Comment by bitinn** [comment]
This issue is addressed in v2.6.0, see readme:

https://github.com/bitinn/node-fetch#custom-agent
